src/ApparentHorizons/YlmSpherepack.cpp

```diff
@@ -689,106 +689,103 @@ void YlmSpherepack::interpolate(
+  const auto& cos_theta = interpolation_info.cos_theta;
 
-  for (size_t i = 0; i < result->size(); ++i) {
-    const auto& intrp_info = interpolation_info[i];
-    const auto& cos_theta = intrp_info.cos_theta;
+  *result = make_with_value<T>(cos_theta, 0.);
 
-    // Clenshaw recurrence for m=0.  Separate because there is no phi
-    // dependence, and there is a factor of 1/2.
-    size_t idx = 0;
-    {
-      double ycn = 0.0;
-      double ycnp1 = 0.0;
-      for (size_t n = n_theta_ - 1; n > 0;
-           --n, ++idx) {  // Loops from n_theta_-1 to 1.
-        double ycnp2 = ycnp1;
-        ycnp1 = ycn;
-        ycn = cos_theta * alpha[idx] * ycnp1 + beta[idx] * ycnp2 +
-              spectral_coefs[a_offset + spectral_stride * index[idx]];
-      }
-      (*result)[i] += 0.5 * intrp_info.pbar_factor[0] *
-                      (beta[idx] * ycnp1 + cos_theta * alpha[idx] * ycn +
-                       spectral_coefs[a_offset + spectral_stride * index[idx]]);
-      ++idx;
+  // Clenshaw recurrence for m=0.  Separate because there is no phi
+  // dependence, and there is a factor of 1/2.
+  size_t idx = 0;
+  {
+    T ycn = make_with_value<T>(cos_theta, 0.);
+    T ycnp1 = make_with_value<T>(cos_theta, 0.);
+    for (size_t n = n_theta_ - 1; n > 0;
+         --n, ++idx) {  // Loops from n_theta_-1 to 1.
+      T ycnp2 = ycnp1;
+      ycnp1 = ycn;
+      ycn = cos_theta * alpha[idx] * ycnp1 + beta[idx] * ycnp2 +
+            spectral_coefs[a_offset + spectral_stride * index[idx]];
     }
-    // Now do recurrence for other m.
-    for (size_t m = 1; m < l1; ++m) {
-      double ycn = 0.0;
-      double ycnp1 = 0.0;
-      double ysn = 0.0;
-      double ysnp1 = 0.0;
-      for (size_t n = n_theta_ - 1; n > m; --n, ++idx) {
-        double ycnp2 = ycnp1;
-        double ysnp2 = ysnp1;
-        ycnp1 = ycn;
-        ysnp1 = ysn;
-        ycn = cos_theta * alpha[idx] * ycnp1 + beta[idx] * ycnp2 +
-              spectral_coefs[a_offset + spectral_stride * index[idx]];
-        ysn = cos_theta * alpha[idx] * ysnp1 + beta[idx] * ysnp2 +
-              spectral_coefs[b_offset + spectral_stride * index[idx]];
-      }
-      const double fc =
-          intrp_info.pbar_factor[m] *
-          (beta[idx] * ycnp1 + cos_theta * alpha[idx] * ycn +
-           spectral_coefs[a_offset + spectral_stride * index[idx]]);
-      const double fs =
-          intrp_info.pbar_factor[m] *
-          (beta[idx] * ysnp1 + cos_theta * alpha[idx] * ysn +
-           spectral_coefs[b_offset + spectral_stride * index[idx]]);
-      (*result)[i] +=
-          fc * intrp_info.cos_m_phi[m] - fs * intrp_info.sin_m_phi[m];
-      ++idx;
+    *result += 0.5 * interpolation_info.pbar_factor[0] *
+               (beta[idx] * ycnp1 + cos_theta * alpha[idx] * ycn +
+                spectral_coefs[a_offset + spectral_stride * index[idx]]);
+    ++idx;
+  }
+  // Now do recurrence for other m.
+  for (size_t m = 1; m < l1; ++m) {
+    T ycn = make_with_value<T>(cos_theta, 0.);
```

Comment:
Again, this is especially bad. You've very likely lost all performance gains `DataVector` could've given you in these nested loops. Here there are enough allocations that you should really be using a `Variables`